src/components/Articles/CompanyProvisions.vue

```diff
@@ -0,0 +1,227 @@
+        </v-btn>
+
+        <!-- More Actions Menu -->
+        <span class="more-actions">
+          <v-menu offset-y>
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" class="more-actions-btn" v-on="on">
+                <v-icon>mdi-menu-down</v-icon>
+              </v-btn>
+            </template>
+            <v-list>
+              <v-list-item class="actions-dropdown_item" @click="isEditing = true">
+                <v-list-item-subtitle>
+                  <v-icon small>mdi-pencil</v-icon>
+                  <span class="ml-1">{{ editLabel }}</span>
+                </v-list-item-subtitle>
+              </v-list-item>
+            </v-list>
+          </v-menu>
+        </span>
+      </v-flex>
+    </v-layout>
+    <v-layout v-else>
+      <v-flex xs3>
+        <label><strong>Pre-existing Company Provisions</strong></label>
+      </v-flex>
+      <v-flex xs9>
+        <v-layout>
+          <v-flex>
+            <p class="instructions-paragraph">
+              Complete this item only if the company has resolved that none of the Pre-existing Company Provisions
+              are to apply to this company (refer to Part 17 and Table 3 of the Regulation under the Business
+              Corporations Act).
+            </p>
+            <v-layout>
+                <v-checkbox
+                  id="cp-checkbox"
+                  class="mt-1"
+                  :color="isInvalid ? checkBoxColor : ''"
+                  v-model="draftProvisionsRemoved"
+                />
+                <span class="checkbox-label" :class="{ 'invalid': isInvalid }">
+                  The company has resolved that none of the Pre-existing Company Provisions are to apply to this company
+                </span>
+            </v-layout>
+          </v-flex>
+        </v-layout>
+        <v-layout pt-10>
+          <v-flex xs12>
+            <div class="action-btns">
+              <v-btn
+                large
+                color="primary"
+                id="company-provisions-done"
+                @click="setCompanyProvisionsDone()"
+              >
+                <span>Done</span>
+              </v-btn>
+              <v-btn
+                large
+                outlined
+                color="primary"
+                id="company-provisions-cancel"
+                @click="cancelCompanyProvisionChange()"
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-flex>
+        </v-layout>
+      </v-flex>
+    </v-layout>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Mixins, Watch } from 'vue-property-decorator'
+import { CommonMixin } from '@/mixins'
+import { ConfirmDialog } from '@/components/dialogs'
+
+@Component({
+  components: {
+    ConfirmDialog
+  }
+})
+export default class CompanyProvisions extends Mixins(CommonMixin) {
+  private draftProvisionsRemoved: boolean = false
```

Comment:
Redundant type declaration (can be inferred from initial value).

(Some linters complain about this. Not ours -- yet -- but it's good practise anyway.)

Reply:
We need to initialize it to be reactive: https://vuejs.org/v2/guide/reactivity.html

Reply:
My point is that some linters (and maybe ours eventually) will complain about redundant type declaration here (you declare it as boolean and then assign a boolean value).

It's not broken atm, but it might eventually break if you don't use simply:
`private draftProvisionsRemoved = false`

Eg, https://stackoverflow.com/questions/46799828/tslint-type-trivially-inferred-why-is-it-bad-practice-to-include-the-type-he

Reply:
Got it. I thought that typescript allowed you to reassign a string for example to that variable if you don't explicitly declare the type. I checked that it's only possible using ```any``` type. Thanks!